### ISSUE TITLE
youtube.com - "login" walls

### ISSUE THREAD

**Issue by krystian3w** [yourduskquibbles/webannoyances#245]
### URL(s) where the issue occurs

`youtube.com`

### Describe the issue

Pop-up with new (?) string "`Sign in to YouTube`".
It may appear at the first visit or after some time.

No have match into your regex:
https://github.com/yourduskquibbles/webannoyances/blob/024c8dca77994fb6dad1d9576efac283ef40379c/filters/modal_filters.txt#L996-L998

### Screenshot(s)

![](https://user-images.githubusercontent.com/35370833/90939844-18ccbc80-e40d-11ea-93d9-efecc266efc2.png) |
--- |

### Versions

-  Chrome 84
-  uBO 1.29.2

### Settings

- stock and this project

### Notes

Maybe more globe range is use: `:has(a[href^="https://support.google.com/youtube/?p=sign_in"])`

`youtube.com##ytd-popup-container > paper-dialog.ytd-popup-container[role="dialog"]:has(a[href^="https://support.google.com/youtube/?p=sign_in"])`

~~Pop-up may use same background as cookie/rules policy:~~

<del>

```
www.youtube.com##body > .opened[opened][style^="z-index"]
```

</del>

**Comment by yourduskquibbles** [comment]
I can't recreate the popup so impossible for me to test right now 😒

Does it appear on first page load or after going to the site a number of times or staying on the site for extended timeframe?

**Comment by krystian3w** [comment]
For my only after first visit on homepage or when open first video/live link. 
May need remove all `YouTube` cookies (pop-up may not use localstorage).

Unable see this on `/user/`&`/channel/` "section". Pop-up should stop direct opened video after ~3 seconds and change video frame into `thumbnail`.

**Comment by yourduskquibbles** [comment]
I still can't recreate on US IP with first visit in temporary container with no preset cookies. Maybe EU only (will test on VPN) or some other A/B test such as browser or OS?

Test link for direct link to video `https://www.youtube.com/watch?v=jXsdKpOWL1U`

**Comment by krystian3w** [comment]
Good links, here also I see:

![obraz](https://user-images.githubusercontent.com/35370833/91325776-e1bb2a00-e7c3-11ea-8588-cf16274f5519.png) | ![obraz](https://user-images.githubusercontent.com/35370833/91326301-77ef5000-e7c4-11ea-8490-2e42c99d8820.png)
--- |---
pl | en (uk)

Now also have this pop-up with Firefox and "containers".

**Comment by yourduskquibbles** [comment]
Are you on Polish IP for both screenshots? Want to see if I can recreate as much of your environment as possible. Glad they are appearing on those test links for you because I can't get them to appear for me on US IP 🤔but at least have video to test on knowing you can see it.

**Comment by krystian3w** [comment]
Yes - both Polish IP, language change in Chrome settings or via glitch with YouTube settings for guest in right top corner:

![obraz](https://user-images.githubusercontent.com/35370833/91327363-b46f7b80-e7c5-11ea-86e4-969ea9342b77.png) |
--- |

The language changes after refreshing and any cookie is generated about "I've already seen a pushy popup about the recommended login" to hide this pop-up via Google Idea.

**Comment by krystian3w** [comment]
`US` no have that pop-up (Miami VPN).

I rule out that it can be hidden by blocking request/resource, because I see after change VPN to Polish localization.

**Comment by yourduskquibbles** [comment]
I just connected to Polish VPN and on first load get the popup!

**Comment by krystian3w** [comment]
IMO only that link is "stable"/"constant":

![obraz](https://user-images.githubusercontent.com/35370833/91328397-2d230780-e7c7-11ea-96ba-eefa22067068.png) |
--- |


```
<a class="yt-simple-endpoint style-scope yt-formatted-string" spellcheck="false" href="https://support.google.com/youtube/?p=sign_in" rel="nofollow" target="_blank" dir="auto">Dowiedz się więcej / Learn more</a>
```

**Comment by yourduskquibbles** [comment]
why not just these 2 cosmetic filters for popup signin prompt?

```
youtube.com##.opened
youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer
```

**Comment by krystian3w** [comment]
`.opened` we have with other name, "ytd-pop-up-overlay", so is no needed.

`yt-upsell-dialog-renderer` maybe is too generic - maybe generate fail similar as Fanboy Annoyance:

https://github.com/easylist/easylist/commit/37908e6126489cb2cdf1d66787e1fb78e34b3421#diff-a020a1b4fa94a782267fe0a762edd58e
```diff
+! Youtube TV promos
+youtube.com##.ytd-primetime-promo-renderer
+youtube.com##ytd-popup-container
```

https://github.com/easylist/easylist/commit/6aa17e9642d2088dae3b29a63931f93da6219fc4#diff-a020a1b4fa94a782267fe0a762edd58e
```diff
-youtube.com##ytd-popup-container
```

**Comment by yourduskquibbles** [comment]
`youtube.com##ytd-popup-container` is the filter for ANY popup functions on youtube, user initiated or loaded automatically so that is way to generic. Thee are old issues from couple years ago on this repository about that.

Could do a `:has-text(foo)` for the upsell but then you don't get matches for other languages. Just given the class name of `upsell` though I think this is always going to be for some purpose to try and get user to subscribe to youtubeTV or go "premium".

**Comment by randomcookiemonster** [comment]
Hello, I'm having the same problem as described in this issue, only for me the login prompt is not shown in a popup. The login prompt is shown instead of the video in the video container.

Here's a screenshot, though I probably shouldn't have cropped it: https://imgur.com/a/4Q6B158

I hope it's alright to put this here instead of creating a new issues since it's essetianlly the same issue.

**Comment by yourduskquibbles** [comment]
@randomcookiemonster I've never seen that login wall before. What setup are you using when you see this? e.g. Is it only on chrome or mobile? In private mode with cookies blocked? More details would be good as I have never seen this one before on standard US IP

Unfortunately, only way for others to help is if it is recreatable.

**Comment by krystian3w** [comment]
I once saw a similar one on blocked materials when a subscription is not paid for YouTuber.

**Comment by randomcookiemonster** [comment]
> 
> 
> @randomcookiemonster I've never seen that login wall before. What setup are you using when you see this? e.g. Is it only on chrome or mobile? In private mode with cookies blocked? More details would be good as I have never seen this one before on standard US IP
> 
> Unfortunately, only way for others to help is if it is recreatable.

I've never seen anything like it before either, It feels like Youtube is testing something new.

My setup:
- Firefox (desktop, win 10) release version (not beta, nightly etc).
- Not using a private window.
- I'm using Ublock Origin, with these filters:
  - uBlock filters
  - uBlock filters – Badware risks
  - uBlock filters – Privacy
  - uBlock filters – Resource abuse
  - uBlock filters – Unbreak
  - EasyList
  - EasyPrivacy
  - Online Malicious URL Blocklist 
  - Fanboy’s Annoyance 
  - uBlock filters – Annoyances 
  - Peter Lowe’s Ad and tracking server list
  - SWE: Frellwit's Swedish Filter
  - Malware Domain List
  - Malware domains

**Comment by randomcookiemonster** [comment]
> 
> 
> I once saw a similar one on blocked materials when a subscription is not paid for YouTuber.

Good idea, but im my case I don't think this is the case. None of the channels I've watched videos from require a subscription. The video will also play after I click "NOT NOW".

**Comment by yourduskquibbles** [comment]
standard residential US IP? No vpn/proxy?

**Comment by randomcookiemonster** [comment]
> 
> 
> standard residential US IP? No vpn/proxy?

Sorry, forgot that part. I'm using a standard Swedish IP, No vpn or proxy.

**Comment by randomcookiemonster** [comment]
A possibly related problem that started at the same time is that, whenever the login prompt isn't shown, videos will start playing, the page reloads after a second, and the video doesn't play automatically this time. Maybe the prompt can't be shown and it forces a page reload?

**Comment by krystian3w** [comment]
> Swedish IP

I can only reproduce to high buttons:

![obraz](https://user-images.githubusercontent.com/35370833/91602889-5761fa00-e96c-11ea-8e51-ed3aa9ee8d7d.png) | ![obraz](https://user-images.githubusercontent.com/35370833/91602861-4ca76500-e96c-11ea-84c9-f6dbb135c9ec.png)
--- | ---
touch version on PC without touch | normal

**Comment by randomcookiemonster** [comment]
> 
> 
> > Swedish IP
> 
> I can only reproduce too high buttons:
> ![obraz](https://user-images.githubusercontent.com/35370833/91602889-5761fa00-e96c-11ea-8e51-ed3aa9ee8d7d.png) 	![obraz](https://user-images.githubusercontent.com/35370833/91602861-4ca76500-e96c-11ea-84c9-f6dbb135c9ec.png)
> touch version on PC without touch 	normal

I'm not sure what you mean, I haven't mentioned any buttons?

**Comment by krystian3w** [comment]
play bar:

![obraz](https://user-images.githubusercontent.com/35370833/91603173-d0f9e800-e96c-11ea-9b52-0fce6ccbaa84.png) | ![obraz](https://user-images.githubusercontent.com/35370833/91603227-eff87a00-e96c-11ea-8a15-ab8b25b90ece.png)
--- | ---
Swedish A/B test | normal

**Comment by yourduskquibbles** [comment]
This combination of network and cosmetic filters blocks the signin wall for me on `https://www.youtube.com/watch?v=jXsdKpOWL1U?t=8` using Swedish VPN

```
||youtube.com/*jsbin/spf.vflset/*$script,domain=youtube.com
||youtube.com/notifications_ajax?action_get_registration_token=1$xhr,domain=youtube.com
||consent.google.com^$subdocument,domain=youtube.com
||google.com/js/bg/$script,domain=youtube.com
||youtube.com/youtubei/v1/log_event$xhr,domain=youtube.com


youtube.com##.opened
youtube.com###consent-bump
youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer
```

@randomcookiemonster and @krystian3w can you test? It also still allowed me to login if i wanted to with `Sign in` button.

**Comment by randomcookiemonster** [comment]
> ```
> ||youtube.com/*jsbin/spf.vflset/*$script,domain=youtube.com
> ||youtube.com/notifications_ajax?action_get_registration_token=1$xhr,domain=youtube.com
> ||consent.google.com^$subdocument,domain=youtube.com
> ||google.com/js/bg/$script,domain=youtube.com
> ||youtube.com/youtubei/v1/log_event$xhr,domain=youtube.com
> 
> 
> youtube.com##.opened
> youtube.com###consent-bump
> youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer
> ```

Thank you I will test this, might take a while to be sure it works.

Do I add everything to "My Filters"?

**Comment by randomcookiemonster** [comment]
> Edit: it doesn't start the video at timestamp requested but does an initial load then a sub reload? video is at least watchable without clicking through popups though?
Sorry I'm not sure what you mean by this?

**Comment by krystian3w** [comment]
> Do I add everything to "My Filters"?

yes.

**Comment by randomcookiemonster** [comment]
> Also I don't understand, filters break e.g. `https://www.youtube.com/watch?v=jXsdKpOWL1U?t=8`

I think that's what he said.

**Comment by randomcookiemonster** [comment]
> 
> 
> > Edit: it doesn't start the video at timestamp requested but does an initial load then a sub reload? video is at least watchable without clicking through popups though?
> 
> Sorry I'm not sure what you mean by this?
> 
> Edit: Oh, the filters reloads the page?

If my assumption is correct, which filter is responsible for discarding the "t"-param (and probably all params?)?

**Comment by randomcookiemonster** [comment]
> ```
> ||youtube.com/*jsbin/spf.vflset/*$script,domain=youtube.com
> ||youtube.com/notifications_ajax?action_get_registration_token=1$xhr,domain=youtube.com
> ||consent.google.com^$subdocument,domain=youtube.com
> ||google.com/js/bg/$script,domain=youtube.com
> ||youtube.com/youtubei/v1/log_event$xhr,domain=youtube.com
> 
> 
> youtube.com##.opened
> youtube.com###consent-bump
> youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer
> ```
> 
> @randomcookiemonster and @krystian3w can you test? It also still allowed me to login if i wanted to with `Sign in` button.
> 
@yourduskquibbles Unfortunately this doesn't stop the login prompt.

**Comment by krystian3w** [comment]
I suppose these break timera from comments or description video: `1:08` / `1:32:15`

**Comment by krystian3w** [comment]
https://github.com/uBlockOrigin/uAssets/issues/7842

**Comment by yourduskquibbles** [comment]
I think the signin popup script code is being loaded through `https://www.youtube.com/s/desktop/5068772e/jsbin/desktop_polymer_inlined_html_polymer_flags.vflset/desktop_polymer_inlined_html_polymer_flags.js`

You can use ctrl + f and search for `Polymer({is:"ytd-consent-bump-renderer"` to find in above source but  I don't know enough js/script defusers to know if it is able to be defused though? 

<details>
<summary>minified code section (click to expand)</summary>

```javascript
Polymer({is:"ytd-consent-bump-renderer",behaviors:[V.YtRendererBehavior,T.YtActionSenderBehavior,X.YtActionHandlerBehavior],screenLayer:4,properties:{data:{type:Object,observer:"maybeForceConsent_"}},actionMap:{"yt-close-popup-action":"handlePopupClose_"},handlePopupClose_:function(a){this.signinPromptClosed_||"yt-upsell-dialog-renderer"!==a||(this.signinPromptClosed_=!0,this.maybeForceConsent_(this.data))},
created:function(){this.expressVe_=this.acceptVe_=this.lightbox_=null;this.signinPromptClosed_=!0;z("kevlar_sign_in_prompt_before_mandatory_consent")&&(this.signinPromptClosed_=!1);this.disableConsentBump_=z("mandatory_consent_bump")&&!qia()},
ready:function(){var a=this;this.disableConsentBump_||z("kevlar_sign_in_prompt_before_mandatory_consent")&&uk(xk,function(){a.forceShowIfNoBlockingSigninPrompt()},5E3)},
attached:function(){this.disableConsentBump_?this.hideConsentBump_():(this.listen(window,"message","onReceivedMessage_"),this.lightbox_&&this.fire("yt-add-element-to-app",this.lightbox_))},
detached:function(){this.disableConsentBump_||this.unlisten(window,"message","onReceivedMessage_")},
maybeForceConsent_:function(a){this.disableConsentBump_||(a&&this.async(function(){this.fire("yt-consent-bump-display-changed")}),this.initializeVE_(),a&&a.forceConsent?(this.hideConsentBump_(),this.signinPromptClosed_&&(this.async(this.pausePlayer_.bind(this),1E3),this.openLightbox_())):"true"==pl("HideTicker","false")&&this.hideConsentBump_())},
initializeVE_:function(){var a=LN();MN(a,106531,void 0,{layer:4,parentLayer:0});this.acceptVe_=RN(a,{veType:104841},void 0,4);this.expressVe_=RN(a,{veType:105964},void 0,4)},
createLightbox_:function(){var a=document.createElement("ytd-consent-bump-lightbox");a.id="consent-bump";document.body.appendChild(a);this.data&&this.data.forceConsent?(a.getDialog().modal=!0,z("mandatory_consent_bump")&&(a.getDialog().style.maxHeight="550px")):a.getDialog().withBackdrop=!0;this.fire("yt-add-element-to-app",a);return a},
getConsentBumpIcon_:function(a){return a.toLowerCase()},
onReceivedMessage_:function(a){if(!this.lightbox_){var b=this.getLightbox_();if(b)this.lightbox_=b;else return}b=Md(a.origin);if(WLa[b]&&WLa.hasOwnProperty(b)||Gd("consent_url_override"))switch((b=kn(this.getScreenLayer()))&&("cb-ui-done-early"===a.data&&this.acceptVe_?$K(b,this.acceptVe_):"cb-express-done"===a.data&&this.expressVe_&&$K(b,this.expressVe_)),a.data){case"cb-ui-done-early":case"cb-ui-done":case"cb-express-done":case"cb-already-consented":this.hideConsentBump_();case"cb-user-closed":this.lightbox_.getDialog().close();
this.resumePlayer_();break;case"verify-origin":a.source.parent==window&&a.source.postMessage("verify-origin-reply",a.origin);}},
onRemindMeLaterTap_:function(){ol("HideTicker","true",86400);this.hideConsentBump_()},
onReviewTap_:function(){this.pausePlayer_();this.openLightbox_()},
openLightbox_:function(){var a=kn(this.getScreenLayer());a&&(this.acceptVe_&&YK(a,[this.acceptVe_]),this.expressVe_&&z("mandatory_consent_bump")&&YK(a,[this.expressVe_]));(a=this.getLightbox_())||(a=this.createLightbox_());a.getIframe().src=this.data.iframeUrl;this.lightbox_=a;a.getDialog().open()},
pausePlayer_:function(){var a=WO(this);a&&a.pause()},
hideConsentBump_:function(){this.hidden=!0;this.fire("yt-consent-bump-display-changed")},
resumePlayer_:function(){var a=WO(this);a&&a.play()},
getLightbox_:function(){var a=document.querySelector("ytd-app")||document.querySelector("ytg-app");return a?a.querySelector("#consent-bump"):null},
forceShowIfNoBlockingSigninPrompt:function(){var a=this.sendAction("yt-get-popup-opened-action",["yt-upsell-dialog-renderer"]);0!==a.length&&a[0]||this.handlePopupClose_("yt-upsell-dialog-renderer")},get _template(){if(!this.__$tmpl){this.__$tmpl=document.createElement("template");this.__$tmpl.innerHTML="<!--css-build:shady-->\n    \n    \n    <yt-icon id=\"consent-shield\" icon=\"yt-icons:[[getConsentBumpIcon_(data.securityIcon.iconType)]]\" class=\"style-scope ytd-consent-bump-renderer\">\n    </yt-icon>\n    <yt-formatted-string id=\"alert-message\" text=\"[[data.alertMessage]]\" class=\"style-scope ytd-consent-bump-renderer\">\n    </yt-formatted-string>\n    <ytd-button-renderer id=\"remind-me-later-button\" data=\"[[data.remindMeLaterButton.buttonRenderer]]\" on-tap=\"onRemindMeLaterTap_\" class=\"style-scope ytd-consent-bump-renderer\">\n    </ytd-button-renderer>\n    <ytd-button-renderer id=\"review-button\" data=\"[[data.reviewButton.buttonRenderer]]\" on-tap=\"onReviewTap_\" class=\"style-scope ytd-consent-bump-renderer\">\n    </ytd-button-renderer>\n  "}return this.__$tmpl},set _template(v){this.__$tmpl=v}});
```
</details>



<details>
<summary>unminified readable format (click to expand)</summary>

```javascript
Polymer({
    is: "ytd-consent-bump-renderer",
    behaviors: [V.YtRendererBehavior, T.YtActionSenderBehavior, X.YtActionHandlerBehavior],
    screenLayer: 4,
    properties: {
        data: {
            type: Object,
            observer: "maybeForceConsent_"
        }
    },
    actionMap: {
        "yt-close-popup-action": "handlePopupClose_"
    },
    handlePopupClose_: function (a) {
        this.signinPromptClosed_ || "yt-upsell-dialog-renderer" !== a || (this.signinPromptClosed_ = !0, this.maybeForceConsent_(this.data))
    },
    created: function () {
        this.expressVe_ = this.acceptVe_ = this.lightbox_ = null;
        this.signinPromptClosed_ = !0;
        z("kevlar_sign_in_prompt_before_mandatory_consent") && (this.signinPromptClosed_ = !1);
        this.disableConsentBump_ = z("mandatory_consent_bump") && !qia()
    },
    ready: function () {
        var a = this;
        this.disableConsentBump_ || z("kevlar_sign_in_prompt_before_mandatory_consent") && uk(xk, function () {
            a.forceShowIfNoBlockingSigninPrompt()
        }, 5E3)
    },
    attached: function () {
        this.disableConsentBump_ ? this.hideConsentBump_() : (this.listen(window, "message", "onReceivedMessage_"), this.lightbox_ && this.fire("yt-add-element-to-app", this.lightbox_))
    },
    detached: function () {
        this.disableConsentBump_ || this.unlisten(window, "message", "onReceivedMessage_")
    },
    maybeForceConsent_: function (a) {
        this.disableConsentBump_ || (a && this.async(function () {
                this.fire("yt-consent-bump-display-changed")
            }), this.initializeVE_(), a && a.forceConsent ? (this.hideConsentBump_(), this.signinPromptClosed_ && (this.async(this.pausePlayer_.bind(this), 1E3), this.openLightbox_())) : "true" == pl("HideTicker", "false") && this.hideConsentBump_())
    },
    initializeVE_: function () {
        var a = LN();
        MN(a, 106531, void 0, {
            layer: 4,
            parentLayer: 0
        });
        this.acceptVe_ = RN(a, {
            veType: 104841
        }, void 0, 4);
        this.expressVe_ = RN(a, {
            veType: 105964
        }, void 0, 4)
    },
    createLightbox_: function () {
        var a = document.createElement("ytd-consent-bump-lightbox");
        a.id = "consent-bump";
        document.body.appendChild(a);
        this.data && this.data.forceConsent ? (a.getDialog().modal = !0, z("mandatory_consent_bump") && (a.getDialog().style.maxHeight = "550px")) : a.getDialog().withBackdrop = !0;
        this.fire("yt-add-element-to-app", a);
        return a
    },
    getConsentBumpIcon_: function (a) {
        return a.toLowerCase()
    },
    onReceivedMessage_: function (a) {
        if (!this.lightbox_) {
            var b = this.getLightbox_();
            if (b)
                this.lightbox_ = b;
            else
                return
        }
        b = Md(a.origin);
        if (WLa[b] && WLa.hasOwnProperty(b) || Gd("consent_url_override"))
            switch ((b = kn(this.getScreenLayer())) && ("cb-ui-done-early" === a.data && this.acceptVe_ ? $K(b, this.acceptVe_) : "cb-express-done" === a.data && this.expressVe_ && $K(b, this.expressVe_)), a.data) {
            case "cb-ui-done-early":
            case "cb-ui-done":
            case "cb-express-done":
            case "cb-already-consented":
                this.hideConsentBump_();
            case "cb-user-closed":
                this.lightbox_.getDialog().close();
                this.resumePlayer_();
                break;
            case "verify-origin":
                a.source.parent == window && a.source.postMessage("verify-origin-reply", a.origin);
            }
    },
    onRemindMeLaterTap_: function () {
        ol("HideTicker", "true", 86400);
        this.hideConsentBump_()
    },
    onReviewTap_: function () {
        this.pausePlayer_();
        this.openLightbox_()
    },
    openLightbox_: function () {
        var a = kn(this.getScreenLayer());
        a && (this.acceptVe_ && YK(a, [this.acceptVe_]), this.expressVe_ && z("mandatory_consent_bump") && YK(a, [this.expressVe_]));
        (a = this.getLightbox_()) || (a = this.createLightbox_());
        a.getIframe().src = this.data.iframeUrl;
        this.lightbox_ = a;
        a.getDialog().open()
    },
    pausePlayer_: function () {
        var a = WO(this);
        a && a.pause()
    },
    hideConsentBump_: function () {
        this.hidden = !0;
        this.fire("yt-consent-bump-display-changed")
    },
    resumePlayer_: function () {
        var a = WO(this);
        a && a.play()
    },
    getLightbox_: function () {
        var a = document.querySelector("ytd-app") || document.querySelector("ytg-app");
        return a ? a.querySelector("#consent-bump") : null
    },
    forceShowIfNoBlockingSigninPrompt: function () {
        var a = this.sendAction("yt-get-popup-opened-action", ["yt-upsell-dialog-renderer"]);
        0 !== a.length && a[0] || this.handlePopupClose_("yt-upsell-dialog-renderer")
    },
    get _template() {
        if (!this.__$tmpl) {
            this.__$tmpl = document.createElement("template");
            this.__$tmpl.innerHTML = "<!--css-build:shady-->\n    \n    \n    <yt-icon id=\"consent-shield\" icon=\"yt-icons:[[getConsentBumpIcon_(data.securityIcon.iconType)]]\" class=\"style-scope ytd-consent-bump-renderer\">\n    </yt-icon>\n    <yt-formatted-string id=\"alert-message\" text=\"[[data.alertMessage]]\" class=\"style-scope ytd-consent-bump-renderer\">\n    </yt-formatted-string>\n    <ytd-button-renderer id=\"remind-me-later-button\" data=\"[[data.remindMeLaterButton.buttonRenderer]]\" on-tap=\"onRemindMeLaterTap_\" class=\"style-scope ytd-consent-bump-renderer\">\n    </ytd-button-renderer>\n    <ytd-button-renderer id=\"review-button\" data=\"[[data.reviewButton.buttonRenderer]]\" on-tap=\"onReviewTap_\" class=\"style-scope ytd-consent-bump-renderer\">\n    </ytd-button-renderer>\n  "
        }
        return this.__$tmpl
    },
    set _template(v) {
        this.__$tmpl = v
    }
});
```
</details>

**Comment by uBlock-user** [comment]
Does adding `youtube.com##ytd-consent-bump-renderer` make a difference ?

**Comment by uBlock-user** [comment]
I was able to reproduce with Polish VPN, `youtube.com###dialog`  works.

**Comment by yourduskquibbles** [comment]
There are two login blocks that I've seen from testing with EU VPNs


1. The Polish IP login wall appears to just be a modal overlay and can be fixed with cosmetic filters e.g. `youtube.com###dialog`

![Polish Wall](https://user-images.githubusercontent.com/35370833/90939844-18ccbc80-e40d-11ea-93d9-efecc266efc2.png)

2. The Swedish IP login wall appears to be more invasive and uses the video player itself to display the login wall, which is why I think a `+js()` type filter would be needed to work around this one. I can't always recreate this wall so it is hard to do a lot of testing on it, but it is the login wall that turns the video player grey reported by @randomcookiemonster [above](https://github.com/yourduskquibbles/webannoyances/issues/245#issuecomment-682655559)

![Swedish Wall](https://user-images.githubusercontent.com/22258847/92133712-b1344b00-edd6-11ea-8c66-71b0131d94cf.png)

**Comment by uBlock-user** [comment]
@yourduskquibbles These work for me

`youtube.com##ytd-popup-container`
`youtube.com###consent-bump`

**Comment by yourduskquibbles** [comment]
`youtube.com##ytd-popup-container` will break all user-initiated clicks that pop open a modal such as `login` & `save`. It's too generic it is how youtube loads all of their automatic and user-generated popup elements.

**Comment by uBlock-user** [comment]
Hmmm, welll I tried with the Swedish IP and I can't reproduce the black overlay over the player.

**Comment by yourduskquibbles** [comment]
> 
> 
> Hmmm, welll I tried with the Swedish IP and I can't reproduce the black overlay over the player.

I don't even think it is an overlay with the Swedish IP wall, it is the `video` player element itself. The page gets redrawn about 2 seconds into a video and that is what gets displayed in place of the video. (Probably as the JS code gets stepped through in the browser) Then once you click not now or sign in, the page gets redrawn again. My guess is that setting some boolean constant that drives the A/B login-required to a `false` would nuke it

**Comment by krystian3w** [comment]
`youtube.com###consent-bump` - GDPR by Google.

**Comment by yourduskquibbles** [comment]
> 
> 
> `youtube.com###consent-bump` - GDPR by Google:
> 
> > or at any time go to `g.co/privacytools`.
> 
> `youtube.com###dialog` - cut only right part of pop-up and possible leftover left oldie "keyhole":
> ![obraz](https://user-images.githubusercontent.com/35370833/92149489-62120880-ee1e-11ea-9e85-aa5120b37fdd.png)

Do just these filters three fix the `Polish Login Wall`?

```
youtube.com##.opened
youtube.com###consent-bump
youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer
```

**Comment by krystian3w** [comment]
Not 3 but 2:

`youtube.com##.opened`
`youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer`



I think that as long as they do not block the rewind (scroll), maybe Fanboy Annoyance should also cut it out.

**Comment by yourduskquibbles** [comment]
So `youtube.com###consent-bump` should **NOT** be added to `modal_filters.txt` list but other two are safe to add?

**Comment by krystian3w** [comment]
If you're see leaking pop-ups about cookies and ad personalization policy (`###consent-bump`), maybe add it to [`cookie_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/cookie_filters.txt).

**Comment by krystian3w** [comment]
And if you are not logged in and want to remove a frozen player, you can try to use it:

`cookie_filters.txt`:
www.youtube.com###consent-bump:remove()

`modal_filters.txt`:
www.youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer:remove()
www.youtube.com##.opened:remove()


I have the impression that at the time, Google is not detecting that the pop-up was and the player is not stopping.

**Comment by yourduskquibbles** [comment]
Marking as partially solved since I have no fix for `Swedish Login Wall`

**Comment by Yaron10** [comment]
I still (intermittently) get the prompt with the following rules:

```
youtube.com##.opened
youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer
```

**Comment by krystian3w** [comment]
Try after update from https://github.com/yourduskquibbles/webannoyances/commit/c384322052e8c0497f38963ccd46defc783e9c5d

this no have filter solution for now: https://i.imgur.com/3NPYoNd.png

**Comment by Yaron10** [comment]
https://i.imgur.com/3NPYoNd.png is the prompt I get.

Thank you.

**Comment by Yaron10** [comment]
It's either https://user-images.githubusercontent.com/10229320/91622121-5649c080-e99e-11ea-8c88-7fdf11c7dbe9.png or https://i.imgur.com/3NPYoNd.png.

If I understand you correctly, there's no solution currently for both.

**Comment by krystian3w** [comment]
Yes, because the tests with the network filters have fail and  that black box over the playback is still visible.

**Comment by Yaron10** [comment]
OK. Thanks again.

**Comment by randomcookiemonster** [comment]
I am now getting both login walls. 😞

I'll try adding:
youtube.com###consent-bump:remove()
youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer:remove()
youtube.com##.opened:remove()
to my filters (uBlock Origin) to if that at least removes the popup.

Thanks for the work you've done on this so far. 👍

**Comment by krystian3w** [comment]
Germany 🇩🇪 also should have pop-up (A/B  test on more people...).

**Comment by krystian3w** [comment]
@randomcookiemonster try this: https://github.com/uBlockOrigin/uAssets/issues/7636#issuecomment-687200509


```js
youtube.com##+js(set, yt_logging_screen, undefined)
```

**Comment by Termuellinator** [comment]
Just found out about this list, thank you, nice thing :+1: 

Regarding Youtube, it has some issues though - it gets rid of the new annoying login-prompt, but the video is then stopped again for me and doesn't autoplay anymore.
The other thing (probably the bigger issue tbh) is that on the YT-Homepage, the sidebar doesn't stay anymore, but scrolls up. And after some scrolling, the video-thumbnails don't show the video-duration anymore. When disabling the list and reloading, all is back to normal there.

**Comment by krystian3w** [comment]
> the video is then stopped again for me and doesn't autoplay anymore.

Race contidion with `:remove()` - that can still stop video if you regullary clear cookies/localStorage.

---

> The other thing (probably the bigger issue tbh) is that on the YT-Homepage, the sidebar doesn't stay anymore, but scrolls up. And after some scrolling, the video-thumbnails don't show the video-duration anymore. When disabling the list and reloading, all is back to normal there.

May need new topic "`youtube - broken menu and miniatures`"

**Comment by randomcookiemonster** [comment]
> I'll try adding:
> youtube.com###consent-bump:remove()
> youtube.com##ytd-popup-container > paper-dialog > yt-upsell-dialog-renderer:remove()
> youtube.com##.opened:remove()
> to my filters (uBlock Origin) to if that at least removes the popup.

This has the side-effect of returnng the issue I described [in an earlier my comment of mine](https://github.com/yourduskquibbles/webannoyances/issues/245#issuecomment-683028179), but I'll sum it up again. The video starts playing, after maybe 1 second the page reloads, and the video is paused at the beginning. I stopped uBlock Origin when this happened, and the popup login wall immediatelly showed up. Ater clicking "No thanks", restarting uBlock Origin and reloading the page the issue that's the topic of this comment was gone. But it will probably return next time the popup login wall is triggered.

**Comment by randomcookiemonster** [comment]
> 
> 
> @randomcookiemonster try this: [uBlockOrigin/uAssets#7636 (comment)](https://github.com/uBlockOrigin/uAssets/issues/7636#issuecomment-687200509)
> 
> ```js
> youtube.com##+js(set, yt_logging_screen, undefined)
> ```

Thanks I will.

**Comment by krystian3w** [comment]
Other user said "scriptlet no block pop-up / black rectangle above video".

**Comment by randomcookiemonster** [comment]
> @randomcookiemonster try this: [uBlockOrigin/uAssets#7636 (comment)](https://github.com/uBlockOrigin/uAssets/issues/7636#issuecomment-687200509)
> ```js
> youtube.com##+js(set, yt_logging_screen, undefined)
> ```
This doesn't not work.

**Comment by Termuellinator** [comment]
I only did some tests, but for me it seems that adding 
```
www.youtube.com###consent-bump
www.youtube.com##.opened
www.youtube.com##ytd-popup-container.ytd-app.style-scope
```
to my uBlock filters gets rid of the login nagscreen and prevents the video from pausing without impacting the yt main page.
(taken from [here](https://www.reddit.com/r/youtube/comments/im18pu/sign_in_spam_suddenly/g41d05q/) )

**Comment by krystian3w** [comment]
`##ytd-popup-container.ytd-app.style-scope` break `notfiy from bell` "🔔" / `subscribe` / `share` as: https://github.com/yourduskquibbles/webannoyances/issues/245#issuecomment-686565977 / https://github.com/yourduskquibbles/webannoyances/issues/245#issuecomment-680988108

Also pop-up is visible once as per time (how long cookies expire) so simple CSS filter may generate FP with fix stop played video.

**Comment by c33s** [comment]
now this also hit austria. can block it with the filters i found on reddit but it stops playing the video :(

because of cookie blocking i would get the login wall on each video

**Comment by krystian3w** [comment]
@c33s try: 

https://github.com/uBlockOrigin/uAssets/issues/7842#issuecomment-694397422

```
youtube.com##+js(set-constant.js, ytInitialPlayerResponse.auxiliaryUi.messageRenderers.upsellDialogRenderer.isVisible, false)
youtube.com##+js(set-constant.js, ytInitialData.topbar.desktopTopbarRenderer.interstitial.consentBumpRenderer.forceConsent, false)
```

**Comment by Takuyax** [comment]
```
www.youtube.com###consent-bump
www.youtube.com##.opened
www.youtube.com##ytd-popup-container.ytd-app.style-scope
```
This worked great for me for a few days, but youtube now interrupts video playback immediately at startup with the attached error message. Reloading removed the error, but it immediately did the previously described auto-reload without autoplay.
No idea if this can even be blocked, since it looks as if they're running some check to see if the nag-screen was displayed.
![Screenshot_20201002_162858](https://user-images.githubusercontent.com/39657731/94935730-b0720000-04cd-11eb-8ef6-b75d2b241da6.png)

**Comment by krystian3w** [comment]
New without close button or hidden by non good filter.

**Comment by krystian3w** [comment]
For some people these may can break search bar: https://github.com/yourduskquibbles/webannoyances/blob/3cd5bf28282cc6724d63c4df23551d4caf361bce/filters/modal_filters.txt#L1012

Maybe better use:

`youtube.com###upsell-dialog-text > a[href*="sign"]:upward(body > * > *)`

**Comment by yourduskquibbles** [comment]
> 
> 
> For some people these may can break search bar:
> 
> https://github.com/yourduskquibbles/webannoyances/blob/3cd5bf28282cc6724d63c4df23551d4caf361bce/filters/modal_filters.txt#L1012
> 
> Maybe better use:
> 
> `youtube.com###upsell-dialog-text > a[href*="sign"]:upward(body > * > *)`

@krystian3w I've updated to your suggestion in https://github.com/yourduskquibbles/webannoyances/commit/80188caad9a20253de997e464362c0e4ea9e2c90

**Comment by krystian3w** [comment]
Possible no longer used by Google.
